src/openforms/utils/json_logic.py

```diff
@@ -49,3 +49,57 @@ def normalize_value(value: Union[dict, list, tuple, int, str]) -> JSONLogicValue
         return JsonLogicTest.from_expression(value)
 
     raise NotImplementedError(f"Unknown value type: {type(value)}")
+
+def find_component_key(test: JsonLogicTest, candidates: list) -> str:
+    for value in test.values:
+        if isinstance(value, JsonLogicTest) and value.operator == "var":
+            candidate = value.values[0]
+            if candidate in candidates:
+                return candidate
+
+        elif isinstance(value, JsonLogicTest) and value.operator == "date":
+            return find_component_key(value, candidates=candidates)
+
+    return ""
+
+
+def find_key_value(value: Union[JsonLogicTest, str]) -> str :
+    """Find the value associated to the key in the json_trigger_rule"""
+    if isinstance(value, JsonLogicTest) :
+        return find_key_value(value.values[0])
+
+    return value
+
+def formating_json_logic(logics :Union[dict, list, JsonLogicTest], candidates, submission):
+    if isinstance(logics, JsonLogicTest):
+        for value in logics.values:
+            if value.operator in ['var', 'date'] :
+                key = find_component_key(logics, candidates)
+                form_step_name, label = '', ''
+                formsteps = submission.form.formstep_set.all()
+                for formstep in formsteps:
+                    if key in formstep.form_definition.get_all_keys() :
+                        label = formstep.form_definition.get_key_label(key)
+                        form_step_name = formstep.form_definition.name
+                        break
+
+                return {
+                    'nodeType' : 'leaf',
+                    'operator': logics.operator,
+                    'key': key,
+                    'value': find_key_value(logics.values[1]),
+                    'stepName': form_step_name,
+                    'label': label
+                }
+            else: return {
+                'nodeType' : 'operation', 
+                'operator' :logics.operator,
+                'children' : formating_json_logic(logics.values, candidates, submission)
+                }
+        return ""
+    
+    else :
+        logic_list = []
+        for logic in logics :
+            logic_list.append(formating_json_logic(logic, candidates, submission))
+        return logic_list
```

Comment:
that doesn't seem right if `logics` is a dict

Reply:
I do think it is impossible that logics is a dict. But maybe you said so because in the signature I was alowing dict (and that was a mistake)